packages/core/lib/commands/call/run.js

```diff
@@ -0,0 +1,167 @@
+    console.log(
+      `WARNING!!! Not a view function\n  ABI stateMutability: ${functionEntry.stateMutability}`
+    );
+  }
+
+  const provider = new Encoder.ProviderAdapter(config.provider);
+  let result;
+  try {
+    result = await provider.request({
+      method: "eth_call",
+      params: [
+        {
+          from: config.from,
+          to: transaction.to,
+          data: transaction.data
+        },
+        config.blockNumber
+      ]
+    });
+  } catch (error) {
+    console.log(`Error Message: ${error.message}\nError Code: ${error.code}`);
+  }
+
+  if (result !== null && result !== undefined) {
+    const [decoding] = await decoder.decodeReturnValue(functionEntry, result);
+
+    for (const { value: result } of decoding.arguments) {
+      console.log(
+        util.inspect(new Codec.Export.ResultInspector(result), {
+          colors: true,
+          depth: null,
+          maxArrayLength: null,
+          breakLength: 79
+        })
+      );
+    }
+  }
+
+  async function sourceFromLocal(contractNameOrAddress, config) {
+    let encoder, decoder;
+    const contractNames = fs
+      .readdirSync(config.contracts_build_directory)
+      .filter(filename => filename.endsWith(".json"))
+      .map(filename => filename.slice(0, -".json".length));
+
+    const contracts = contractNames
+      .map(contractName => ({
+        [contractName]: config.resolver.require(contractName)
+      }))
+      .reduce((a, b) => ({ ...a, ...b }), {});
+
+    const isEmpty = Object.keys(contracts).length === 0;
+    if (isEmpty) {
+      throw new Error(
+        "No artifacts found! Please run `truffle compile` first to compile your contracts"
+      );
+    }
+
+    if (contractNameOrAddress.startsWith("0x")) {
+      const projectInfo = {
+        artifacts: Object.values(contracts)
+      };
+
+      ({ encoder, decoder } = await getEncoderDecoderForContractAddress(
+        contractNameOrAddress,
+        projectInfo
+      ));
+    } else {
+      const settings = {
+        provider: config.provider,
+        projectInfo: {
+          artifacts: Object.values(contracts)
+        }
+      };
+
+      const contract = contracts[contractNameOrAddress];
+      // Error handling to remind users to run truffle migrate first
+      const instance = await contract.deployed();
+      encoder = await Encoder.forContractInstance(instance, settings);
+      decoder = await Decoder.forContractInstance(instance, settings);
+    }
+    return { encoder, decoder };
+  }
+
+  async function sourceFromExternal(contractNameOrAddress, config) {
+    const { compileResult } = await fetchAndCompile(
+      contractNameOrAddress,
+      config
+    );
+
+    const projectInfo = {
+      commonCompilations: compileResult.compilations
+    };
+
+    const { encoder, decoder } = await getEncoderDecoderForContractAddress(
+      contractNameOrAddress,
+      projectInfo
+    );
+
+    return { encoder, decoder };
```

Comment:
Really minor style note, but you can tighten this up with just `return await ...`. :)  Why create a variable when you're just about to return it anyway? :)